corehq/apps/reports/tasks.py

```diff
@@ -80,6 +87,99 @@ def update_calculated_properties_for_domains(domains):
     datadog_report_user_stats('commcare.active_mobile_workers.count', active_users_by_domain)
 
 
+def get_domains_to_update_es_filter():
+    """
+    Returns ES filter to obtain domains that are active, and meet one or more
+    of the following criteria:
+     - never had calculated properties updated
+     - calculated properties was updated over one week ago
+     - new form submissions within the last day
+    """
+    last_week = datetime.utcnow() - timedelta(days=7)
+    more_than_a_week_ago = filters.date_range('cp_last_updated', lt=last_week)
+    not_updated = filters.missing('cp_last_updated')
+    domains_submitted_today = (FormES().submitted(gte=datetime.utcnow() - timedelta(days=1))
+        .terms_aggregation('domain.exact', 'domain').size(0).run().aggregations.domain.keys)
+    is_domain_active = filters.term('is_active', True)
+    return filters.AND(
+        is_domain_active,
+        filters.OR(
+            not_updated,
+            more_than_a_week_ago,
+            filters.term('name', domains_submitted_today)
+        )
+    )
+
+
+@periodic_task(run_every=crontab(hour="22", minute="0", day_of_week="*"), queue='background_queue')
+def update_domain_metrics():
+    domains = get_domains_to_update()
+    for chunk in chunked(domains, 5000):
+        update_domain_metrics_for_domains.delay(chunk)
+
+
+@task(queue='background_queue')
+def update_domain_metrics_for_domains(domains):
```

Comment:
I think we had briefly talked about it awhile back, but can't remember what the decision was on if we should move this task to the data_analytics app? I would be in favor of that for the same reasons we added the SQL model to that app (it just fits better and reports is already overloaded). What do you think?

Reply:
Yes, I think you're correct. I was planning to do that in the follow-up PR that would also remove `calculated_properties`. For now I put it here to make it clear it's essentially the "same" thing being handled differently. Does it make more sense to just put it in the `data_analytics` app now instead, though?

Reply:
Ah sure. I can't think of any real reasons why it would matter one way or the other. A minor thing is the datadog metric for this task would change after that followup PR 🤷🏻‍♂️. So slightly simplifies things to just move it now, plus then the PR to remove calculated properties won't involve moving "unrelated" code.

Reply:
f942db6